File: ios-client/BlockProxy/Models/TunnelStatus.swift
import Foundation

enum TunnelStatus: String, Codable {
    case disconnected
    case connecting
    case connected
    case reconnecting
    case occupied
    case authFailed

    var displayText: String {
        switch self {
        case .disconnected:  return "已断开"
        case .connecting:    return "正在连接..."
        case .connected:     return "已连接"
        case .reconnecting:  return "重连中"
        case .occupied:      return "端口被占用"
        case .authFailed:    return "认证失败"
        }
    }

    var colorName: String {
        switch self {
        case .disconnected:  return "gray"
        case .connecting:    return "yellow"
        case .connected:     return "green"
        case .reconnecting:  return "orange"
        case .occupied:      return "red"
        case .authFailed:    return "red"
        }
    }
}

File: android-client/app/src/main/java/com/blockproxy/android/status/TunnelStatus.kt
package com.blockproxy.android.status

/**
 * Tunnel connection status states, each with a user-visible Chinese display text.
 *
 * The status transitions are managed by the tunnel service; this enum only
 * models the state values and their presentation labels.
 */
enum class TunnelStatus(val displayText: String) {
    Disconnected("已断开"),
    Preparing("准备中"),
    Connecting("正在连接..."),
    Connected("已连接"),
    Reconnecting("重连中"),
    Occupied("端口被占用"),
    AuthFailed("认证失败"),
    Error("错误"),
}
